sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -225,7 +225,7 @@ case class FilterEstimation(plan: Filter) extends Logging {
       attr: Attribute,
       isNull: Boolean,
       update: Boolean): Option[Double] = {
-    if (!colStatsMap.contains(attr) || !colStatsMap(attr).hasCountStats) {
+    if (!colStatsMap.contains(attr) || colStatsMap(attr).nullCount.isEmpty) {
```

Comment:
This fix seems fine, but is that the possible state where `distinctCount.isDefined` is false and `nullCount.isDefined` is true?

Reply:
Ideally this method doesn't require `distinctCount`. Union operator doesn't return `distinctCount`, but it can return `nullCount`.

Reply:
Original commit which added this method was checking only first condition. After refactoring in this commit https://github.com/apache/spark/commit/8077bb04f350fd35df83ef896135c0672dc3f7b0 , the second condition also came. But logically the second condition need not check distinct as it is irrelevant in this method.